src/conditions/TwoRobotBallPlacement.cpp

```diff
@@ -22,7 +22,7 @@ bt::Node::Status TwoRobotBallPlacement::onUpdate() {
     int minimumRequiredRobotsInField = robotDealer::RobotDealer::keeperExistsInWorld() ? 3 : 2;
     bool weHaveEnoughRobots = us.size() >= minimumRequiredRobotsInField;
     //TODO: THIS REMOVES TWOROBOTBALLPLACEMENT (15.1)
-    bool ballIsCloseToBallPlacementPos = ballPlacementPos.dist(ball->pos) < 15.1;
+    bool ballIsCloseToBallPlacementPos = ballPlacementPos.dist(ball->getPos()) < 15.1;
```

Comment:
Although these are issues I would argue they are not relevant to this refactor (Haico did not make any changes except for ball class related things here), and should not be done here since they are clearly seperate issues. 

(Note this specific line is a robocup hack, see the TODO).